### PR TITLE
Fix #1191: Lake Merge Process Step-2 - GQL and GQL Dependencies

### LINKED ISSUE #1191: [Lake] Merge Process Step-2: GQLDataFactory and its dependencies
### Merge Main Functionalities Related to the GQL Data Factory

**In this phase, we will merge the main functionalities related to the GQL Data Factory, ensuring that the GQL-related components are integrated. This includes:**

#### Task List

1. **GQL Data Factory and Their Dependencies**
   - Merge the core GQL Data Factory components.
   - Ensure all dependencies for the GraphQL Data Factory are included and correctly referenced.

2. **Table Classes for GQL**
   - Move table classes designed to work with the GraphQL Data Factory.
   
3. **Changes in the PPSS Folder**
   - Update and add necessary components in the PPSS folder.

4. **Changes in the Subgraph Folder**
   - Modify subgraph components to align with the new GraphQL functionalities.

5. **System Tests**
   - Update system tests to cover new GraphQL functionalities.
   - Ensure all existing tests pass 
 
6. **Changes on the CLI for GQL**
   - Update the Command Line Interface (CLI) to support new GraphQL functionalities.
   - Verify that all CLI commands work correctly and are documented properly.

#### Detailed Steps

1. **Integrate GQL Data Factory Components**
   - [x] Review and merge the GQL components and their dependencies.

2. **Update Table Classes**
   - [x] Merge specific table classes designed for GQL.
   - [x] Perform tests to verify the functionality and compatibility of these table classes.

3. **Revise PPSS Folder**
   - [x] Updates and additions to the PPSS folder. (For only GQL-Data Factory related parts)

4. **Modify Subgraph Components**
   - [x] Update components in the subgraph folder.
   - [x] Run tests to ensure all components are working properly.

5. **Revise System Tests**
   - [x] Update existing system tests to include new GQL functionalities.
   - [x] Ensure all tests pass successfully.

6. **Update CLI for GQL**
   - [x] Modify the CLI to accommodate new GQL functionalities.
   - [x] Test all CLI commands (manual)

### PR DESCRIPTION
**Pull request by kdetry**
Fixes #1191

Changes proposed in this PR:

- The new version of `gql_data_factory` is moved to this branch.
- In the `lake` folder, `etl.py` is removed because it was not compatible with the new `gql_data_factory`.
- Additional tools (`lake_info.py`, `lake_validate.py`) are moved.
- Changes in PPSS are moved to this branch, and all related components are updated because property names have changed.
- The new `table_pdr_predictions.py` is picked from the DuckDB PR. Both `table.py` and `table_pdr_predictions.py` are moved to the `pdr_backend/lake/legacy/` folder.
- CLI modules are modified: ETL commands are removed, and new commands from the DuckDB PR are added.
- The accuracy module is using the legacy subgraph to keep it consistent.

### CODE REVIEW

**Comment by calina-c** [comment]
@kdetry I tried to help a bit with tests and linters. I left a # TODO to encourage a pylint failure where things are pending.

**Comment by KatunaNorbert** [comment]
`pdr get_predictions_info` and `pdr get_predictoors_info` are failing with the following error
Screenshot 2024-06-13 at 17 49 23

**Comment by KatunaNorbert** [comment]
```
pdr lake PPSS_FILE NETWORK
pdr analytics PPSS_FILE NETWORK
```
This commands are also don't work, although they appear in the command lists.
I see that lake cli commands are also added but none of them is working.
This might be related to removing ETL but question is should we merge them this way then, might make more sense to have step 2 and 3 merged together so we don't submit code that is not working or partially working.

**Comment by kdetry** [comment]
CLI command tips are updated, it guide the user now
Prediction commands should be used if the lake has the raw data, it uses the DB. the error message added if there is no data

File: pdr_backend/cli/cli_arguments.py [hunk 1]
```diff
@@ -39,20 +39,21 @@
...
+pdr lake raw drop ppss.yaml sapphire-mainnet 2023-06-01
 """
 
 HELP_OTHER_TOOLS = """
 Power tools:
   pdr multisim PPSS_FILE
   pdr arima_plots PPSS_FILE [--debug_mode False]
   pdr deployer (for >1 predictoor bots)
-  pdr lake PPSS_FILE NETWORK
+  pdr lake LAKE_TYPE(raw, etl) ACTION(drop, update) PPSS_FILE NETWORK
```
Comment (KatunaNorbert) [inline pdr_backend/cli/cli_arguments.py:51]: could you add each command separately, would be easier to understand. Also the drop action has an extra parameter which is the date from which the data should be delete from, and it's not visible here.

I see there are some some examples in the HELP_DOT section but I think the tools sections should be as easy to understand as possible, and also similar with other commands

File: pdr_backend/cli/cli_arguments.py [hunk 2]
```diff
@@ -39,20 +39,21 @@
...
   pdr analytics PPSS_FILE NETWORK
 
 Utilities:
   pdr get_predictoors_info ST END PQDIR PPSS_FILE NETWORK --PDRS
-  pdr get_predictions_info ST END PQDIR PPSS_FILE NETWORK --FEEDS
+  pdr    ST END PQDIR PPSS_FILE NETWORK --FEEDS
```
Comment (KatunaNorbert) [inline pdr_backend/cli/cli_arguments.py:56]: I think this wasn't intended to be removed

Comment (kdetry) [reply pdr_backend/cli/cli_arguments.py]: Sorry, my mistake

File: pdr_backend/cli/cli_arguments.py (hunk 1 above)
Comment (kdetry) [reply pdr_backend/cli/cli_arguments.py]: I changed it now 👍

**Pull request merged**